inst/include/Rcpp/date_datetime/Date.h

```diff
@@ -127,7 +127,7 @@ namespace Rcpp {
         // update m_tm based on m_d
         void update_tm() {
             if (R_FINITE(m_d)) {
-                time_t t = 24*60*60 * m_d;      // (fractional) days since epoch to seconds since epoch
+                time_t t = (time_t)(24*60*60 * m_d);      // (fractional) days since epoch to seconds since epoch
```

Comment:
I think you may need to use a `static_cast<time_t>()` in place of the c-style / regular cast.